### PR TITLE
Making prod_type field required in the Model

### PR DESCRIPTION
**Pull request by steeve85**
This is a fix for https://github.com/DefectDojo/django-DefectDojo/issues/2278

TLDR: Using the API it was possible to create a **Product** without a **Product Type** which was resulting in server error when the "All engagements" page (`/engagements_all`) was accessed.

Making the `prod_type` field required in the model as suggested by @valentijnscholten fix this issue (and potentially other similar issues).

**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR 

Current accepted labels for PRs:
- bugfix

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Thanks. Is there now an appropriate error message when a product your is not supplied via the api?

**Comment by steeve85** [comment]
@valentijnscholten Yes there is an appropriate error message (which is the default message implemented in the API): _{"prod_type":["This field is required."]}_

`$ curl -X POST "http://localhost:8080/api/v2/products/" -H "accept: application/json" -H "Authorization: Token XXX" -H "Content-Type: application/json" -H "X-CSRFToken: XXX" -d "{ \"name\": \"test missing product_type\", \"description\": \"test desc\"}" -D -`

HTTP response:

> HTTP/1.1 400 Bad Request
> Server: nginx/1.17.10
> Date: Thu, 21 May 2020 03:49:44 GMT
> Content-Type: application/json
> Content-Length: 41
> Connection: keep-alive
> X-Content-Type-Options: nosniff
> X-Frame-Options: SAMEORIGIN
> X-XSS-Protection: 1; mode=block
> Vary: Cookie
> Allow: GET, POST, HEAD, OPTIONS
> 
> {"prod_type":["This field is required."]}

**Comment by steeve85** [comment]
When I run `python3 manage.py makemigrations dojo` I'm having the following error

> SystemCheckError: System check identified some issues:
> 
> ERRORS:
> defectDojo_engagement_survey.Answered_Survey.assignee: (fields.E304) Reverse accessor for 'Answered_Survey.assignee' clashes with reverse accessor for 'Answered_Survey.assignee'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.assignee' or 'Answered_Survey.assignee'.
> defectDojo_engagement_survey.Answered_Survey.assignee: (fields.E305) Reverse query name for 'Answered_Survey.assignee' clashes with reverse query name for 'Answered_Survey.assignee'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.assignee' or 'Answered_Survey.assignee'.
> defectDojo_engagement_survey.Answered_Survey.responder: (fields.E304) Reverse accessor for 'Answered_Survey.responder' clashes with reverse accessor for 'Answered_Survey.responder'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.responder' or 'Answered_Survey.responder'.
> defectDojo_engagement_survey.Answered_Survey.responder: (fields.E305) Reverse query name for 'Answered_Survey.responder' clashes with reverse query name for 'Answered_Survey.responder'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.responder' or 'Answered_Survey.responder'.
> dojo.Answered_Survey.assignee: (fields.E304) Reverse accessor for 'Answered_Survey.assignee' clashes with reverse accessor for 'Answered_Survey.assignee'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.assignee' or 'Answered_Survey.assignee'.
> dojo.Answered_Survey.assignee: (fields.E305) Reverse query name for 'Answered_Survey.assignee' clashes with reverse query name for 'Answered_Survey.assignee'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.assignee' or 'Answered_Survey.assignee'.
> dojo.Answered_Survey.responder: (fields.E304) Reverse accessor for 'Answered_Survey.responder' clashes with reverse accessor for 'Answered_Survey.responder'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.responder' or 'Answered_Survey.responder'.
> dojo.Answered_Survey.responder: (fields.E305) Reverse query name for 'Answered_Survey.responder' clashes with reverse query name for 'Answered_Survey.responder'.
> 	HINT: Add or change a related_name argument to the definition for 'Answered_Survey.responder' or 'Answered_Survey.responder'.

Any idea how I can debug that?

**Comment by valentijnscholten** [comment]
I had these too, but running the migrations made them dissapear. I think the survey / questionaire migration solves it and/or you need to remove the old surveys app from settings.py.

**Comment by valentijnscholten** [comment]
Rebased + updated migration to avoid too many iterations and administrative work on rebasing in the future. Now with 2 approvals, I can merge it.
